src/analysis/ref_mode.rs

```diff
@@ -25,7 +26,7 @@ impl RefMode {
             ..
         }) = env.config.objects.get(&tid.full_name(library))
         {
-            if direction == library::ParameterDirection::In {
+            if direction == library::ParameterDirection::In && transfer != library::Transfer::Full {
                 return ref_mode;
             } else {
                 return RefMode::None;
```

Comment:
I've been thinking about removing this entirely and letting the code below handle it, but everything that's supposed to be `transfer-full` turns back into `&mut` references. Isn't `transfer-full` handled elsewhere, too, or is it okay to pass these as borrow in the first place??

Reply:
`transfer full` and passing by reference has no relationship really. We pass everything by reference, always. It's only GStreamer's miniobjects that need special handling because the reference count there is connected to the mutability (like in Rust's `Arc`).

Reply:
Then this indeed does not make much sense.